standard/types.md

```diff
@@ -1042,4 +1044,123 @@ A compiler may use any expression that dereferences a variable, property, or eve
 >
 > *end example*
 
+#### 8.9.5.2 Type conversions
+
+A compiler that generates diagnostic warnings conforms to these rules.
+
+> *Note:* Differences in top-level or nested nullability annotations in types do not affect whether conversion between the types is permitted, since there is no semantic difference between a non-nullable reference type and its corresponding nullable type ([§8.9.1](types.md#891-general)).
+
+A compiler may issue a warning when nullability annotations differ between two types, either top-level or nested, when the conversion is narrowing.
+
+> *Example*: Types differing in top-level annotations
+>
+> <!-- Example: {template:"code-in-class-lib", name:"TopLevelNullabilityConversionWarnings"} -->
+> ```csharp
+> #nullable enable
+> public class C
+> {
+>     public void M1(string p)
+>     {
+>         _ = (string?)p; // No warning, widening
+>     }
+>
+>     public void M2(string? p)
+>     {
+>         _ = (string)p; // Warning, narrowing
+>         _ = (string)p!; // No warning, suppressed
+>     }
+> }
+> ```
+>
+> *end example*
+
```

Comment:
This Markdown violation occurs often in the spec. Just disable and re-enable the rule. You'll find lots of examples in the standard.

Reply:
I might be misunderstanding. This seems to be rule <https://github.com/DavidAnson/markdownlint/blob/v0.37.4/doc/md028.md>. Scrolling through files and a regex search in all files for `^>.*\r\n\r\n>` didn't turn anything up, so I haven't found any examples to follow.

Looking at what this rule is, it seems to be saving people from creating block quotes that get unified by some parsers and not others. If that's not a concern in this spot, then it seems that it must not be a concern for the entire repo, and maybe we could ignore MD028 globally with the run of this tool. What do you think?

Reply:
Oh, 😆 Of course that search will find nothing, because there are intervening lines that disable and reenable the rule. I got it now.

I'd still be interested in knowing whether we could turn the rule off for the whole repo. Perhaps inserting the HTML comments actually addresses MD028 in its own way :D